### PR TITLE
Components: FontSizePicker: Adjust Select Button sizing

### LINKED ISSUE #19472: Drop downs for text settings too tall
This is what I see on paragraph settings in Twenty Twenty:

image

Here it is in Twenty Nineteen:

image

### PR DESCRIPTION
**Pull request by ItsJonQ**
## Description
This update enhances `CustomSelectControl` to accept props to modify it's
internal `Button` component (`Object`). Since `FontSizePicker` uses
`CustomSelectControl`, this update allows us to specify the `isSmall`
property for the `Button`. By doing so, it fixes the alignment of the Select
with the other form control elements.

## How has this been tested?
This has been tested locally in Gutenberg dev.

## Screenshots

Before:
71903431-fd51c180-315b-11ea-95f3-94bab6bbd05f

After:
Screen Shot 2020-01-07 at 10 48 45 AM

## Types of changes
Adding a new prop to `CustomSelectControl` to be extended within `FontSizePicker`

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR. .

Resolves: https://github.com/WordPress/gutenberg/issues/19472

### CODE REVIEW

File: packages/components/src/custom-select-control/README.md [hunk 1]
```diff
@@ -102,6 +102,12 @@ Function called with the control's internal state changes. The `selectedItem` pr
 - Type: `Function`
 - Required: No
 
+#### selectButtonProps
```
Comment (mtias) [inline packages/components/src/custom-select-control/README.md]: I think this could just be called `isSmall` directly.

**Comment by youknowriad** [comment]
Actually I wonder if we need prop at all here. For me the `CustomSelectControl` is a select, so its height should follow inputs, selects.... and not buttons.

Ultimately, I believe form controls also can see their default height increased but this requires a larger effort so can we just make is small by default?

**Comment by ItsJonQ** [comment]
@youknowriad + @mtias Thank you for your feedback! 🙏 

I've removed `selectButtonProps`, and opted for `isSmall` with a default of `true` as suggested. My original intention was to provide the ability to customize the rendering of the internal `Button` if required.

> CustomSelectControl is a select, so its height should follow inputs, selects.... and not buttons.

I feel like the 3 (input, select, button) are all related in terms of visuals. They're often combined together, so they have similar heights.

>  but this requires a larger effort so can we just make is small by default?

I agree :)

File: packages/components/src/custom-select-control/index.js [hunk 2]
```diff
@@ -51,6 +51,7 @@ export default function CustomSelectControl( {
 	label,
 	options: items,
 	onChange: onSelectedItemChange,
+	isSmall = true,
```
Comment (youknowriad) [inline packages/components/src/custom-select-control/index.js]: To be fair,  I'm still not sure we should even expose this on the CustomSelectControl as it's just a select and SelectControl, for instance, doesn't have that option.

**Review by youknowriad — APPROVED** [review]
Thanks for the fix 👍

**Comment by youknowriad** [comment]
Rebased the branch and pushed some small tweaks in an attempt to merge. I hope you don't mind :)

**Pull request merged**